### PR TITLE
feat: add diode

### PR DESCRIPTION
**Pull request by kjappelbaum**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kjappelbaum** [comment]
@mrzv are you ok with being listed as maintainer?

**Comment by mrzv** [comment]
Sure. That's fine with me.

**Comment by kjappelbaum** [comment]
>@kjappelbaum The recipe looks good to me, except: you don't want to list yourself as a maintainer? Or is that automatic since you are contributing the PR?

I'm fine with also being a maintainer, I just didn't want to push myself to the fore for no good reasons ;)

**Comment by kjappelbaum** [comment]
@conda-forge/help-python, ready for review!

File: recipes/diode/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/diode-{{ version }}.tar.gz
+  sha256: bb863e036fa2c3a9ba19db1c462629abbaffbc731f20b524e9924f22c384b08e
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [win]
+  entry_points:
```
Comment (carterbox) [inline recipes/diode/meta.yaml:17]: Empty entry points?

**Review by carterbox — CHANGES_REQUESTED** [review]
This package is vending pybind11. Need an explanation why revending is  needed or use latest pybind from conda-forge.

And pybind license should be included because it is statically linked.


https://conda-forge.org/docs/maintainer/knowledge_base.html#pybind11-abi-constraints

**Comment by kjappelbaum** [comment]
@mrzv I didn't realize that `diode` vendors `pybind11` - what's the reason for that?

**Comment by mrzv** [comment]
pybind's own instructions and examples using CMake assume that pybind11 is vendored with the code. If you have an example of using pybind11 with CMake without vendoring, I'm happy to consider it (but the code would need to build not only with conda, but also from pypi).

Fixing the version (by vendoring) also helps to avoid incompatibility bugs down the line. And I don't see many downsides.

File: recipes/diode/meta.yaml (hunk 1 above)
Comment (kjappelbaum) [reply recipes/diode/meta.yaml:17]: yes

**Comment by kjappelbaum** [comment]
@conda-forge/help-python-c thanks for your feedback. I included the pybind11 license. diode's maintainer listed above the reasons for vendoring pybind11:

>pybind's own instructions and examples using CMake assume that pybind11 is vendored with the code. If you have an example of using pybind11 with CMake without vendoring, I'm happy to consider it (but the code would need to build not only with conda, but also from pypi).

> Fixing the version (by vendoring) also helps to avoid incompatibility bugs down the line. And I don't see many downsides.

File: recipes/diode/meta.yaml (hunk 1 above)
Comment (carterbox) [reply recipes/diode/meta.yaml:17]: Why?

Comment (carterbox) [reply recipes/diode/meta.yaml:17]: Why? This is very unusual. Please write a comment in the recipe to explain why it is needed. This recipe is hosted publicly and we get a lot of copypasta from people creating recipes from recipes without knowing how they work.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/diode/meta.yaml (hunk 1 above)
Comment (kjappelbaum) [reply recipes/diode/meta.yaml:17]: Thanks for your feedback. Perhaps I don't know how they work. But I don't see any command line script or anything like that being defined by diode (https://github.com/mrzv/diode/blob/e77e9cea8f1be8e167dfcc8a6ee1770378df4859/setup.py#L59-L89).

File: recipes/diode/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+
```
[the remark this answers is not in the document]
Comment (kjappelbaum) [reply recipes/diode/meta.yaml]: thanks!

**Pull request merged**